src/umode.S

```diff
@@ -0,0 +1,161 @@
+// Copyright (c) 2022 by Rivos Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
```

Comment:
do you have to save floating point and vector registers?

Reply:
There's been some discussion about that. Floating point is disabled at this point, so umode -- which should be really seen as a a form of lesser hypervisor code -- doesn't use it. Same for vector.
In general we might enable them in the future, and then yes, we'd have to restore and save them on entry and exit.